### PR TITLE
Skip Flatfield For Non-Image Exposure Types

### LINKED ISSUE #363: Add a check for exposure type for the FlatField Step
_Issue [RCAL-253](https://jira.stsci.edu/browse/RCAL-253) was created on JIRA by [David Davis](https://jira.stsci.edu/secure/ViewProfile.jspa?name=ddavis):_

The flat field step code should include a check for the exposure.type attribute and only perform the flat fielding if exposure.type="WFI_IMAGE". For the Prism and Grism data the code should set cal_step.flat_field="SKIPPED" and exit without running the flat fielding code.

### PR DESCRIPTION
**Pull request by PaulHuwe**
Closes #363 

Resolves RCAL-253 Add a check for exposure type for the FlatField Step

**Description**

This PR adds a check for WFI_Image in flat field step, and skips the step (with appropriate calstep update) if a different exposure type. Test of this is also added.

Checklist
- [x ] Tests

- [ ] Documentation

- [x ] Change log

- [x ] Milestone

- [ ] Label(s)

### CODE REVIEW

File: romancal/flatfield/flat_field.py [hunk 1]
```diff
@@ -54,6 +54,11 @@ def do_flat_field(output_model, flat_model):
                     'shape as the science data')
         log.warning('Step will be skipped')
         output_model.meta.cal_step.flat_field = 'SKIPPED'
+    elif output_model.meta.exposure.type != "WFI_IMAGE":
+        # Check to see if attempt to flatten non-Image data
+        log.warning('Exposure type is '+output_model.meta.exposure.type+', which is not WFI_IMAGE')
```
Comment (ddavis-stsci) [inline romancal/flatfield/flat_field.py:59]: I'd suggest log.info instead of log warning since skipping the step is not an error.

**Review by ddavis-stsci — APPROVED** [review]
Looks good

File: romancal/flatfield/flat_field.py (hunk 1 above)
Comment (nden) [reply romancal/flatfield/flat_field.py:59]: I agree. I would make it one line log message, e.g.
`log.info("Skipping flat field for spectral exposure."`

**Review by nden — APPROVED** [review]

**Review by dmggh — APPROVED** [review]
Not that it matters in a practical sense, but checking if flat fielding works might be more convincing if the flat fielding values are not all 1.0 (unlike at least 1 test here).

File: romancal/flatfield/flat_field.py (hunk 1 above)
Comment (PaulHuwe) [reply romancal/flatfield/flat_field.py:59]: Good catch! Changed.

File: CHANGES.rst [hunk 2]
```diff
@@ -11,6 +11,11 @@ assign_wcs
 
 - Added ``assign_wcs`` step to romancal. [#361]
 
+flat
+----
+
+- Added check for WFI_Image in flat field step (skip if a different exposure type). Added test. [#366]
```
[the remark this answers is not in the document]
Comment (PaulHuwe) [reply CHANGES.rst:17]: Changed.

**Pull request merged**